docs/current_docs/api/clients-cli.mdx

```diff
@@ -0,0 +1,32 @@
+---
+slug: /api/clients/cli
+---
+
+# Dagger CLI
+
+The Dagger CLI offers a `dagger query` sub-command, which provides an easy way to call the Dagger API from the command line.
+
+This example demonstrates how to build a Go application by cloning the [canonical Git repository for Go](https://go.googlesource.com/example/+/HEAD/hello) and building the "Hello, world" example program from it by calling the Dagger API via `dagger query`.
+
+Create a new shell script named `build.sh` and add the following code to it:
+
+```shell file=./snippets/clients-cli/build.sh
```

Comment:
@marcosnils when I run this I see ```dagger run ./build.sh 
Setup tracing at https://dagger.cloud/traces/setup. To hide: export NOTHANKS=1

✔ connect 1.2s

Error: make request: Post "http://dagger/query": unexpected EOF
Error: make request: Post "http://dagger/query": read tcp 127.0.0.1:46514->127.0.0.1:41499: read: connection reset by peer
Build unsuccessful```

Can you please help update the code?

Reply:
@vikram-dagger here's the updated script: 

```
#!/bin/bash

# get Go examples source code repository
source=$(dagger query <<EOF | jq -r .git.branch.tree.id
{
  git(url:"https://go.googlesource.com/example") {
    branch(name:"master") {
      tree {
        id
      }
    }
  }
}
EOF
)

# mount source code repository in golang container
# build Go binary
# export binary from container to host filesystem
build=$(dagger query <<EOF | jq -r .container.from.withDirectory.withWorkdir.withExec.file.export
{
  container {
    from(address:"golang:latest") {
      withDirectory(path:"/src", directory:"$source") {
        withWorkdir(path:"/src/hello") {
          withExec(args:["go", "build", "-o", "dagger-builds-hello", "."]) {
            file(path:"./dagger-builds-hello") {
              export(path:"./dagger-builds-hello")
            }
          }
        }
      }
    }
  }
}
EOF
)

if [ -n "$build" ]; then
	echo "Build successful"
else
	echo "Build unsuccessful"
fi
```

You can call it with `./build.sh` (without dagger run)

Reply:
Thanks, updated! But why do you prefer not using `dagger run`? Isn't wrapping in `dagger run` the recommended approach, because it provides live progress? This is why we recommend `dagger run go ...`, `dagger run python ...`, etc.